### PR TITLE
refactor: Refactor negative test cases in refund.test to use Jest assertions 

### PR DESCRIPTION
**Pull request by Deepakchowdavarapu**
Refactor Negative Test Cases in refund.test.ts to Use Jest Assertions

Replaced try-catch blocks with Jest’s .rejects matcher for error scenarios
in refund.test.ts. Updated test cases to use .toMatch and .toThrow for
error assertions, improving test readability and maintainability.

Test results remain consistent, and all assertions now leverage Jest’s
built-in methods.

- Rebased onto upstream/main
- Squashed into a single commit
- Signed-off commit

BREAKING CHANGE: None

Signed-off-by: Deepakchowdavarapu deepak.23bcs10092@ms.sst.scaler.com

### CODE REVIEW

**Comment by Deepakchowdavarapu** [comment]
@petermetz  @outSH please review this !!

File: extensions/cactus-plugin-htlc-coordinator-besu/src/test/typescript/integration/plugin-htlc-coordinator/refund.test.ts [hunk 1]
```diff
@@ -312,22 +312,24 @@ describe("HTLC Coordinator Besu", () => {
...
+    data: {
+      cause: {
+        receipt: {
+          revertReason: expect.stringMatching(/0e494e56414c49445f5345435245540/)
+        }
       }
     }
+  }
+}));
+
```
Comment (outSH) [inline extensions/cactus-plugin-htlc-coordinator-besu/src/test/typescript/integration/plugin-htlc-coordinator/refund.test.ts:332]: Fix the indentation to match the rest of the file

**Review by outSH — CHANGES_REQUESTED** [review]
Thanks for the updates. In the future don't open a separate PR but contact us on discord so we can assist you with any git problems you may have :) 

I've added a single commnet and I'd like you to fix the title as mentioned in previous PR, and read the contributing guide:
- Fix the title, use the one peter proposed in related issue (https://github.com/hyperledger/cacti/issues/3457).

File: extensions/cactus-plugin-htlc-coordinator-besu/src/test/typescript/integration/plugin-htlc-coordinator/refund.test.ts [hunk 2]
```diff
@@ -312,22 +312,24 @@ describe("HTLC Coordinator Besu", () => {
...
-        log.debug("Caught response: %o", exp.response.data);
-        const revertReason = exp.response.data.cause.receipt.revertReason;
-        const regExp = new RegExp(/0e494e56414c49445f5345435245540/);
-        expect(revertReason).toMatch(regExp);
-        // t.match(revertReason, regExp, rejectMsg);
-      } else {
-        throw exp;
+await expect(
+  htlcCoordinatorBesuApiClient.withdrawCounterpartyV1(withdrawCounterparty)
+).rejects.toThrowErrorMatchingSnapshot();
```
Comment (petermetz) [inline extensions/cactus-plugin-htlc-coordinator-besu/src/test/typescript/integration/plugin-htlc-coordinator/refund.test.ts:317]: @Deepakchowdavarapu What does this additional assertion accomplish? Is this covering some edge case scenario that the other assertion does not?

**Review by petermetz — CHANGES_REQUESTED** [review]
@Deepakchowdavarapu Plus one to everything @outSH asked for.

**Comment by petermetz** [comment]
Closing this pull request due to inactivity, but please feel free to reopen as needed and then we can continue the review.
